examples/hybrid/schur_complement_W.jl

```diff
@@ -206,3 +198,82 @@ function linsolve!(::Type{Val{:init}}, f, u0; kwargs...)
         end
     end
 end
+
+"""
+    thomas_algorithm!(A, b)
+
+Thomas algorithm for solving a linear system A x = b,
+where A is a tri-diagonal matrix.
+A and b are overwritten.
+Solution is written to b
+"""
+function thomas_algorithm!(A, b)
+    # A[_, 1] is lower diag, A[_, 2] is main diag, A[_, 3] is upper diag
+    nrows = size(A, 1)
+    # first row
+    A[1, 3] /= A[1, 2]
+    b[1] /= A[1, 2]
+    # interior rows
+    for row in 2:(nrows - 1)
+        fac = A[row, 2] - (A[row, 1] * A[row - 1, 3])
+        A[row, 3] /= fac
+        b[row] = (b[row] - A[row, 1] * b[row - 1]) / fac
+    end
+    # last row
+    fac = A[nrows, 2] - A[nrows - 1, 3] * A[nrows, 1]
+    b[nrows] = (b[nrows] - A[nrows, 1] * b[nrows - 1]) / fac
+    # back substitution
+    for row in (nrows - 1):-1:1
+        b[row] -= b[row + 1] * A[row, 3]
+    end
+    return nothing
+end
+
+function column_thomas_solve!(::ClimaComms.CPUSingleThreaded, A, b)
+    Ni, Nj, _, _, Nh = size(Spaces.local_geometry_data(axes(A)))
+    for h in 1:Nh, j in 1:Nj, i in 1:Ni
```

Comment:
Is there a reason we're not using `bycolumn` here? It would allow us to combine this and the multithreaded implementations